4_processing_features/extract_sc_features_dp.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Process single cell morphology features for DeepProfiler readouts
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+import pandas as pd
+
+from pycytominer import feature_select
+from pycytominer.cyto_utils import DeepProfiler_processing, output, infer_cp_features
+
+
+# ## Set paths to DP project
+
+# In[2]:
+
+
+dp_output_nuc = pathlib.Path("../3_extracting_features/NF1_nuc_project-DP")
+dp_output_cyto = pathlib.Path("../3_extracting_features/NF1_cyto_project-DP")
+
+index_file_nuc = pathlib.Path(f"{dp_output_nuc}/inputs/metadata/index.csv")
+index_file_cyto = pathlib.Path(f"{dp_output_cyto}/inputs/metadata/index.csv")
+
+
+profile_dir_nuc = pathlib.Path(f"{dp_output_nuc}/outputs/efn_pretrained/features")
+profile_dir_cyto = pathlib.Path(f"{dp_output_cyto}/outputs/efn_pretrained/features")
+
+
+# ## Set paths to outputs
+
+# In[3]:
+
+
+output_dir = pathlib.Path('data')
+
+output_file_raw_nuc = pathlib.Path(f'{output_dir}/nf1_sc_deepprofiler_nuc.csv.gz')
+output_file_raw_cyto = pathlib.Path(f'{output_dir}/nf1_sc_deepprofiler_cyto.csv.gz')
+
+output_file_norm_nuc = pathlib.Path(f'{output_dir}/nf1_sc_norm_deepprofiler_nuc.csv.gz')
+output_file_norm_cyto = pathlib.Path(f'{output_dir}/nf1_sc_norm_deepprofiler_cyto.csv.gz')
+
+output_file_norm_fs_nuc = pathlib.Path(f'{output_dir}/nf1_sc_norm_fs_deepprofiler_nuc.csv.gz')
+output_file_norm_fs_cyto = pathlib.Path(f'{output_dir}/nf1_sc_norm_fs_deepprofiler_cyto.csv.gz')
+
+
+# ## Perform normalization and feature selection on DP Nuclei project
+
+# ### Create DeepProfilerData object
+
+# In[4]:
+
+
+deep_data_nuc = DeepProfiler_processing.DeepProfilerData(
+    index_file_nuc, profile_dir_nuc, filename_delimiter="/", file_extension=".npz"
+)
+
+
+# ### Initalize SingleCellDeepProfiler class
+
+# In[5]:
+
+
+deep_single_cell_nuc = DeepProfiler_processing.SingleCellDeepProfiler(deep_data_nuc)
+
+
+# ### Compile raw single cell data
+
+# In[6]:
+
+
+nuc_sc = deep_single_cell_nuc.get_single_cells(output=True, location_x_col_index=1, location_y_col_index=2)
+output(nuc_sc, output_file_raw_nuc)
+
+print(nuc_sc.shape)
+nuc_sc.head()
+
+
+# ### Normalize raw single cell data
+
+# In[7]:
+
+
+normalized_nuc = deep_single_cell_nuc.normalize_deep_single_cells(
+    output_file=output_file_norm_nuc, location_x_col_index=1, location_y_col_index=2
+)
+
+print(normalized_nuc.shape)
+normalized_nuc.head()
+
+
+# ### Separate metadata and features prior to feature selection
+
+# In[8]:
+
+
+# extract metadata prior to feature selection
+metadata_cols = infer_cp_features(normalized_nuc, metadata=True)
+derived_features = [
+    x for x in normalized_nuc.columns.tolist() if x not in metadata_cols
+]
+
+
+# ### Feature selection from normalized data
+
+# In[9]:
+
+
+feature_select_ops = [
+    "variance_threshold",
+    "correlation_threshold",
+    "blocklist",
```

Comment:
Blocklist will only work with CP features. You can remove!

Reply:
(Please remove in both nuc and cyto)

Reply:
👍 Done!